### PR TITLE
Fix deltas of flattened buildings should be flat #824

### PR DESCRIPTION
**Pull request by Med-H**
# Flattening should work when delta is active

_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Issue: #824 

## Description

**Descriptive pull request text**, answering:
  - Buildings do not become flat when delta is active
  - When adding the buildings, the height should only add the deltas after checking if it's flat property is not set. The delta color of the building has to be set accordingly when the building is flattened. Some small refactoring have been done to the variables. 

## Screenshots or gifs
flattenBuildings

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 
 ### Fixed 🐞
 
+-   Buildings are flattened when delta is active.
```
Comment (ce-bo) [inline CHANGELOG.md:18]:
```suggestion
-   Buildings are flattened when delta is active #824.
```

File: visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts [hunk 2]
```diff
@@ -121,15 +121,20 @@ export class CodeMapMesh {
 	}
 
 	private setNewDeltaColor(building: CodeMapBuilding, state: State) {
-		if (building.node.deltas) {
-			const deltaValue = building.node.deltas[state.dynamicSettings.heightMetric]
+		const { appSettings : {mapColors}, dynamicSettings : {heightMetric} } = state;
+		const { node } = building;
+		
+		if (node.flat)
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts:127]: I think, we should not omit the brackets. What do you think?

**Review by ce-bo — COMMENTED** [review]
Would you please add a unit/regression test?

File: visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts (hunk 2 above)
Comment (BridgeAR) [reply visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts:127]: This should likely be fixed by running prettier. Seems like the pre-commit hook was not executed (we still have that, right?).

File: CHANGELOG.md (hunk 1 above)
Comment (BridgeAR) [reply CHANGELOG.md:18]: We might want to use full links as these are not converted to a link in the document itself.

```suggestion
-   Buildings are flattened when delta is active [#824](https://github.com/MaibornWolff/codecharta/issues/824).
```

File: .gitignore [hunk 3]
```diff
@@ -32,6 +32,7 @@ gradle-app.setting
 
 # IDEs and editors
 /.idea
+/.vscode
```
Comment (BridgeAR) [inline .gitignore:35]: Just a general note: it's a good practice to add things like these to an own global ignore file (https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/).

File: CHANGELOG.md (hunk 1 above)
Comment (Med-H) [reply CHANGELOG.md:18]: I will try it out :)

File: .gitignore (hunk 3 above)
Comment (Med-H) [reply .gitignore:35]: I don't get it, could you give me more hints?

**Comment by Med-H** [comment]
> Would you please add a unit/regression test?



> Would you please add a unit/regression test?

will try it out

File: visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts (hunk 2 above)
Comment (RomanenkoVladimir) [reply visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.ts:127]: Yes we do, however theneed to be enabled if you dont commit via commandline

File: visualization/app/codeCharta/ui/codeMap/rendering/geometryGenerator.spec.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	let testNodes: Node[]
+	
+
+	beforeEach(() => {
+		mockStoreService()
+		initData()
+	})
+
+	const mockStoreService = () => {
+		STATE.dynamicSettings.heightMetric = "a" // set to a, since it is the delta defined in TEST_NODES
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/codeMap/rendering/geometryGenerator.spec.ts]: Nice comment, thanks!

File: visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.spec.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		
+		it("should set flat color when flat", () => {
+			setFlattened(true)
+			rebuildMesh()
+
+			expect(TEST_NODE_ROOT.flat).toBeTruthy()
+			expect(codeMapBuilding.deltaColor).toEqual(mapColors.flat)
+		})
+	})
+})
```
Comment (ce-bo) [inline visualization/app/codeCharta/ui/codeMap/rendering/codeMapMesh.spec.ts:50]: Please add a new line at the end of the file :-)

**Review by ce-bo — APPROVED** [review]
Looking Good To Me (LGTM).
Would you please add a new line of the end of a file? I have added two remarks for that.

**Pull request merged**